jerry-debugger/jerry_client_ws.py

```diff
@@ -926,253 +926,254 @@ def print_source(self, line_num, offset):
         return msg
 
 
-# pylint: disable=too-many-branches,too-many-locals,too-many-statements
-def _parse_source(debugger, data):
-    source_code = ""
-    source_code_name = ""
-    function_name = ""
-    stack = [{"line": 1,
-              "column": 1,
-              "name": "",
-              "lines": [],
-              "offsets": []}]
-    new_function_list = {}
+    # pylint: disable=too-many-branches,too-many-locals,too-many-statements
+    def parse_source(self, data):
```

Comment:
If these functions are not to be called from outside of the class, then python coding guidelines suggest prefixing them with either one or two underscores, making them 'protected' or 'private' (aka weakly and strongly internal), respectively. Right now, this patch is actually exposing these functions rather than hiding them (it is turning protected module-level functions into public class methods).

Reply:
so these should be `def _parse_source(self, data)` ? Is this a guideline or enforced in some way?

Reply:
Yes, this is better called either `_parse_source` or `__parse_source`. If it is a non-public API of the class. These guidelines come from [PEP8](https://www.python.org/dev/peps/pep-0008/), _the_ style guide for python. (As most things in Python, however, they are not enforced, only suggested. But the proper use of visibility is hard to enforce in any language, actually.)

Some excerpts from PEP8:
> * _single_leading_underscore: weak "internal use" indicator. E.g. from M import * does not import objects whose name starts with an underscore.
> * __double_leading_underscore: when naming a class attribute, invokes name mangling (inside class FooBar, __boo becomes _FooBar__boo; see below).
> * Public attributes should have no leading underscores.